app/services/file-manager.ts

```diff
@@ -144,6 +144,7 @@ export class FileManagerService extends Service {
       await this.writeFile(filePath, file.data);
 
       if (version !== file.version) {
+        // TODO: throw here seems to be used as control flow
```

Comment:
Kind of.  This is a tough one to get right.  If we wrote an out of date file it's a catastrophic failure the same as if the write call above had failed.  I'm not saying this couldn't be rewritten to avoid the throw here, but I doubt the end result would be more readable/maintainable.